playbooks/gateway-basic-backup.yml

```diff
@@ -1,9 +1,20 @@
 ---
 # This playbook will backup Gateway configurations and store them on the Ansible controller. 
 # See the readme under roles/gateway_basic_backup for setup instructions.
-
+- name: Detect source gateway mysql nodes
+  hosts: gateway_mysql
```

Comment:
Perhaps this first task should be to build the src_host using the gateway_mysql and gateway_processing_node groups.
The second task remains the same.  It will use the src_host and apply the role.